pkg/client/cache/delta_fifo.go

```diff
@@ -452,6 +456,27 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will send a sync event for each item
+func (f *DeltaFIFO) Resync() error {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+	for _, k := range f.knownObjects.ListKeys() {
```

Comment:
Actually, as @rrati observed offline, we should probably iterate over f.items instead.
knownObjects seem to contain objects that were already removed, and for those we don't want to do resync.